### ISSUE TITLE
Make host lowercasing consistent

### ISSUE THREAD

**Issue by nvie** [gruns/furl#70]
There's a minor discrepancy in how the hostname is lowercased, based on which API you use.

##### Steps to replicate

This seems OK:

```python
>>> url = furl('https://MyHostWithCaps.example.org')
>>> url.host
'myhostwithcaps.example.org'
>>> str(url)
'https://myhostwithcaps.example.org'
```

This seems not OK (or at least not consistent):

```python
>>> url = furl('https://MyHostWithCaps.example.org')
>>> url.host = 'MyHostWithCaps.example.org'             # <- difference here!
>>> url.host
'MyHostWithCaps.example.org'
>>> str(url)
'https://MyHostWithCaps.example.org'
```

##### Expected result

The `u.host` property should _always_ return a lowercased value.

##### Actual result

The `u.host` property can _sometimes_ return a string with non-lowercase chars.

**Comment by gruns** [comment]
Wonderful catch.

Fixed in furl v0.4.95.

```python
>>> import furl
>>> furl.__version__
'0.4.95'
>>> from furl import furl
>>> f = furl('https://MyHostWithCaps.example.org')
>>> f.url
'https://myhostwithcaps.example.org'
>>> f.host = 'MyHostWithCaps.example.org'
>>> f.url
'https://myhostwithcaps.example.org'
```

Upgrade with `pip install furl -U`.

Thank you for the thorough, detailed Issue and bringing this bug to my
attention, Vincent. Don't hesitate to let me know if there's anything else I can
do for you.

**Comment by nvie** [comment]
Thanks for the fix!
